src/ChainRouter.php

```diff
@@ -30,7 +29,7 @@
  * @author Henrik Bjornskov <henrik@bjrnskov.dk>
  * @author Magnus Nordlander <magnus@e-butik.se>
  */
-class ChainRouter implements ChainRouterInterface, WarmableInterface
+class ChainRouter extends ChainRouterGenerateBcLayer implements ChainRouterInterface, WarmableInterface
```

Comment:
we could have the if/else for the ChainRouter, and have the ChainRouterBcLayer provide the protected methods and all, but have the actual signature difference in this class. that would eliminate one level of nested inheritance.

Reply:
Wouldn't it be better to keep this construction so that we an just remove the bc layer class as we drop sf <5 support? Original class stays the way it is and the generate method is moved back and the bc class is removed. Or am I misunderstanding your comment?

Reply:
i think both have up- and downsides (the diff is probably more readable with what i suggested), but there is less fiddling around with the final class with your approach.

you are doing the work, i let you chose ;-)

Reply:
I'm personally more in favor of the current setup, is a bit more clear that all the bc stuff is in the separate class and it is easier to clean up afterwards